test/index.js

```diff
@@ -0,0 +1,450 @@
+      assert(eql({ foo: 'baz' }, { foo: 'baz' }), 'eql({ foo: "baz" }, { foo: "baz" })');
+    });
+
+    it('returns true for deeply nested objects', function () {
+      assert(eql({ foo: { bar: 'foo' } }, { foo: { bar: 'foo' } }),
+        'eql({ foo: { bar: "foo" }}, { foo: { bar: "foo" }})');
+    });
+
+    it('returns true with objects with same circular reference', function () {
+      var objectA = { foo: 1 };
+      var objectB = { foo: 1 };
+      var objectC = { a: objectA, b: objectB };
+      objectA.bar = objectC;
+      objectB.bar = objectC;
+      assert(eql(objectA, objectB) === true,
+        'eql({ foo: 1, bar: objectC }, { foo: 1, bar: objectC }) === true');
+    });
+
+    it('returns false with objects containing different literals', function () {
+      assert(eql({ foo: 1, bar: 1 }, { foo: 1, bar: 2 }) === false,
+        'eql({ foo: 1, bar: 2 }, { foo: 1, bar: 2 }) === false');
+      assert(eql({ foo: 'bar' }, { foo: 'baz' }) === false, 'eql({ foo: "bar" }, { foo: "baz" }) === false');
+      assert(eql({ foo: { bar: 'foo' } }, { foo: { bar: 'baz' } }) === false,
+        'eql({ foo: { bar: "foo" }}, { foo: { bar: "baz" }}) === false');
+    });
+
+    it('returns false with objects containing different keys', function () {
+      assert(eql({ foo: 1, bar: 1 }, { foo: 1, baz: 2 }) === false,
+        'eql({ foo: 1, bar: 2 }, { foo: 1, baz: 2 }) === false');
+      assert(eql({ foo: 'bar' }, { bar: 'baz' }) === false, 'eql({ foo: "bar" }, { foo: "baz" }) === false');
+    });
+
+    it('returns false with recursive objects of differing values', function () {
+      var objectA = { foo: 1 };
+      var objectB = { foo: 1 };
+      objectA.bar = objectB;
+      objectB.bar = objectA;
+      assert(eql(objectA, objectB) === false,
+        'eql({ foo: 1, bar: -> }, { foo: 1, bar: <- }) === false');
+    });
+
+  });
+
+  describe('functions', function () {
+
+    it('returns true for same functions', function () {
+      function foo() {}
+      assert(eql(foo, foo), 'eql(function foo() {}, function foo() {})');
+    });
+
+    it('returns false for different functions', function () {
+      assert(eql(function foo() {}, function bar() {}) === false,
+        'eql(function foo() {}, function bar() {}) === false');
+    });
+
+  });
+
+  describe('errors', function () {
+
+    it('returns true for same errors', function () {
+      var error = new Error('foo');
+      assert(eql(error, error), 'eql(error, error)');
+    });
+
+    it('returns false for different errors', function () {
+      assert(eql(new Error('foo'), new Error('foo')) === false,
+        'eql(new Error("foo"), new Error("foo")) === false');
+    });
+
+  });
+
+  describe('@@iterator Sham', function () {
+    before(function () {
+      var globalObject = typeof window === 'undefined' ? global : window;
+      globalObject.Symbol = globalObject.Symbol || {};
+      if (!Symbol.iterator) {
+        Symbol.iterator = '__@@iterator__';
+      }
```

Comment:
I'm not a huge fan of shamming for the purposes of tests, would prefer to test wide range of versions with CI to ensure the library works under different conditions. But if we must, then I'd say that this test needs to completely clean up after itself via an `after`.